File: 1646-kth-missing-positive-number/1646-kth-missing-positive-number.js
/**
 * @param {number[]} arr
 * @param {number} k
 * @return {number}
 */
var findKthPositive = function(arr, k) {
    let st = 0, end = arr.length - 1;
    while (st <= end) {
        let mid = st + Math.floor((end - st) / 2);
        let missing = arr[mid] - (mid + 1);
        if (missing < k) {
            st = mid + 1;
        } else {
            end = mid - 1;
        }
    }
    return st + k;
};

File: 1646-kth-missing-positive-number/1646-kth-missing-positive-number.py
class Solution:
    def findKthPositive(self, arr: List[int], k: int) -> int:
        st, end = 0, len(arr) - 1
        while st <= end:
            mid = st + (end - st) // 2
            missing = arr[mid] - (mid + 1)
            if missing < k:
                st = mid + 1
            else:
                end = mid - 1
        return st + k
